llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -98,3 +98,119 @@ int RISCVTTIImpl::getIntImmCostIntrin(Intrinsic::ID IID, unsigned Idx,
 bool RISCVTTIImpl::shouldFavorPostInc() const {
   return ST->hasExtXCoreVMem();
 }
+
+bool RISCVTTIImpl::isHardwareLoopProfitable(Loop *L, ScalarEvolution &SE,
+                                            AssumptionCache &AC,
+                                            TargetLibraryInfo *LibInfo,
+                                            HardwareLoopInfo &HWLoopInfo) {
+  if (!ST->hasExtXCoreVHwlp()) {
+    return false;
+  }
```

Comment:
NIT usually those are written without braces:

```suggestion
  if (!ST->hasExtXCoreVHwlp())
    return false;
```

Same with the once below.